core/arch/riscv/mm/core_mmu_arch.c

```diff
@@ -745,7 +763,7 @@ bool core_mmu_find_table(struct mmu_partition *prtn, vaddr_t va,
 		pgt = core_mmu_xlat_table_entry_pa2va(pte, pgt);
 		if (!pgt)
 			goto out;
-		va_base += SHIFT_U64(idx, CORE_MMU_SHIFT_OF_LEVEL(level));
+		va_base += core_mmu_pgt_get_va_base(level, idx);
```

Comment:
Does this work as intended?
The previous code adds the va index offset in the translation table, but this new code does something different for non-RV32 code, if I understand it correctly.

Reply:
Actually, this fixes the cases that index >= 256:

```shell
Root table index 000 ==> va_base = 0x000000000000000
Root table index 001 ==> va_base = 0x000008000000000
......
Root table index 255 ==> va_base = 0x0007F8000000000
Root table index 256 ==> va_base = 0xFFF800000000000 (bit[63:48] all equal to bit 47 which is 1)
Root table index 257 ==> va_base = 0xFFF808000000000 (bit[63:48] all equal to bit 47 which is 1)
......
Root table index 511 ==> va_base = 0xFFFFF8000000000 (bit[63:48] all equal to bit 47 which is 1)
```

Adds the va index offset in the translation table is not enough.
As you can see, when the index >= 256, the upper bit[63:48] become 1.
Because bit[47] needs to be extended to bit[63:48] in RISC-V Sv48.
This is what `core_mmu_pgt_get_va_base()` does.

Reply:
I see that, but what happens on the next iteration when the top bits of `va_base` are already `1`?

Reply:
Got it, thanks.